src/download.ts

```diff
@@ -51,22 +56,33 @@ export function getSkylinkUrl(this: SkynetClient, skylink: string, customOptions
   const opts = { ...defaultDownloadOptions, ...this.customOptions, ...customOptions };
   const query = opts.download ? { attachment: true } : {};
 
-  const url = makeUrl(this.portalUrl, opts.endpointPath, parseSkylink(skylink));
+  skylink = parseSkylink(skylink);
+  if (opts.subdomain) {
+    skylink = convertSkylinkToBase32(skylink);
+  }
+  const url = opts.subdomain
+    ? addSubdomain(this.portalUrl, skylink)
+    : makeUrl(this.portalUrl, opts.endpointPath, skylink);
   return addUrlQuery(url, query);
 }
 
 export function getHnsUrl(this: SkynetClient, domain: string, customOptions: any = {}): string {
   const opts = { ...defaultDownloadHnsOptions, ...this.customOptions, ...customOptions };
   const query = opts.download ? { attachment: true } : {};
 
-  const url = makeUrl(this.portalUrl, opts.endpointPath, trimUriPrefix(domain, uriHandshakePrefix));
+  domain = trimUriPrefix(domain, uriHandshakePrefix);
+  const url = opts.subdomain
+    ? addSubdomain(addSubdomain(this.portalUrl, opts.hnsSubdomain), domain)
+    : makeUrl(this.portalUrl, opts.endpointPath, domain);
   return addUrlQuery(url, query);
 }
 
 export function getHnsresUrl(this: SkynetClient, domain: string, customOptions: any = {}): string {
   const opts = { ...defaultResolveHnsOptions, ...this.customOptions, ...customOptions };
 
-  return makeUrl(this.portalUrl, opts.endpointPath, trimUriPrefix(domain, uriHandshakeResolverPrefix));
+  domain = trimUriPrefix(domain, uriHandshakeResolverPrefix);
+  const url = makeUrl(this.portalUrl, opts.endpointPath, domain);
+  return url;
```

Comment:
Let's revert to what was here before - I don't think any changes here are necessary

```suggestion
  return makeUrl(this.portalUrl, opts.endpointPath, trimUriPrefix(domain, uriHandshakeResolverPrefix));
```